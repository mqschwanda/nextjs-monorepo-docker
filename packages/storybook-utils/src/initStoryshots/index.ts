import type { StoryshotsOptions } from '@storybook/addon-storyshots/dist/api/StoryshotsOptions';
import initAddonStoryshots from '@storybook/addon-storyshots';

/**
 * Options for the `initStoryshots` function.
 */
export type InitStoryshotsOptions = StoryshotsOptions & {
  /**
   * If the snapshot tests fail when react logs a key error.
   *
   * Set this to `false` if stories avoid using the key because it would show up in the code example.
   *
   * @default true
   */
  failOnReactKeyError?: boolean
};

/**
 * A wrapper for the `@storybook/addon-storyshots` module
 *
 * See [addon storyshots docs](https://storybook.js.org/addons/@storybook/addon-storyshots) for more information.
 */
export default function initStoryshots({
  failOnReactKeyError = true,
  ...options
}: InitStoryshotsOptions) {
  if (failOnReactKeyError === false) {
    const consoleError = jest.spyOn(console, 'error');

    beforeAll(() => {
      consoleError.mockImplementation((...args) => {
        const [message] = args;
        if (message.includes('Each child in a list should have a unique "key" prop.')) {
          return;
        }

        console.error(...args); // eslint-disable-line no-console
      });
    });

    afterAll(() => {
      consoleError.mockRestore();
    });
  }

  initAddonStoryshots(options);
}
