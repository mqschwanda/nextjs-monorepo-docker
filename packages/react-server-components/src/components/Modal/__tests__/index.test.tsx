import { render } from '@testing-library/react';
import initStoryshots from '@/__tests__/initStoryshots';
import {
  Modal,
  ModalButtonClose,
  ModalBody,
} from '@';

initStoryshots({
  storyKindRegex: /^@mqs\/react-server-components\/components\/Modal$/,
});

describe('@mqs/react-server-components', () => {
  describe('components', () => {
    describe('<Modal />', () => {
      it('renders', () => {
        const testId = Modal.name;
        const children = 'children...';

        const { getByTestId, asFragment } = render((
          <Modal
            className='test'
          >
            <ModalBody>
              <ModalButtonClose />
              <p>
                { children }
              </p>
            </ModalBody>
          </Modal>
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
