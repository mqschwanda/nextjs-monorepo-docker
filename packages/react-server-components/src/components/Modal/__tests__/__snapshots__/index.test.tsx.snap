// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mqs/react-server-components components <Modal /> renders 1`] = `
<DocumentFragment>
  <dialog
    class="modal test"
    data-testid="Modal"
  >
    <form
      class="modal-box"
      data-testid="ModalBody"
      method="dialog"
    >
      <button
        class="btn btn-ghost btn-circle btn-sm absolute right-2 top-2"
        data-testid="ModalButtonClose"
        type="submit"
      >
        <svg
          class="fill-none stroke-current stroke-1 h-4 w-4"
          data-testid="IconClose"
          fill="currentColor"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18L18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
      <p>
        children...
      </p>
    </form>
    <form
      class="modal-backdrop"
      method="dialog"
    >
      <button
        type="submit"
      >
        close
      </button>
    </form>
  </dialog>
</DocumentFragment>
`;

exports[`Storyshots @mqs/react-server-components/components/Modal Modal Example 1`] = `
<dialog
  className="modal"
  data-testid="Modal"
>
  <form
    className="modal-box"
    data-testid="ModalBody"
    method="dialog"
  >
    <button
      className="btn btn-ghost btn-circle btn-sm absolute right-2 top-2"
      data-testid="ModalButtonClose"
      disabled={false}
      type="submit"
    >
      <svg
        className="fill-none stroke-current stroke-1 h-4 w-4"
        data-testid="IconClose"
        fill="currentColor"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 18L18 6M6 6l12 12"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </button>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut mollis.
  </form>
  <form
    className="modal-backdrop"
    method="dialog"
  >
    <button
      type="submit"
    >
      close
    </button>
  </form>
</dialog>
`;
